lib/md5.c

```diff
@@ -103,8 +88,8 @@ void _sasl_hmac_md5_init(HMAC_MD5_CTX *hmac,
    */
 
   /* start out by storing key in pads */
-  MD5_memset((POINTER)k_ipad, '\0', sizeof k_ipad);
-  MD5_memset((POINTER)k_opad, '\0', sizeof k_opad);
+  memset(k_ipad, '\0', sizeof k_ipad);
+  memset(k_opad, '\0', sizeof k_opad);
```

Comment:
memset will be optimized out in many compilers, you may want consider using explicit_bzero() where supported (glibc 2.25 and later).

Reply:
I understand, but the problem is that you are introducing a regression here.

All modern compilers have code elision detection that is applied to memset because it is a well known function, so it has much higher chances of being elided than a custom function like MD5_memset.

So while I think we should remove MD5_memset I also think we should try to use explict_bzero which is guaranteed to not be elided.

The whole point of the memsets at the end of the function is to make sure keys are not left in memory, but code elision would leave them there.

Reply:
I have come up with the portable solution to use OPENSSL_cleanse.

Reply:
LGTM